x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/ForkingOperatorTestCase.java

```diff
@@ -353,4 +354,26 @@ public void setThreadPool() {
     public void shutdownThreadPool() {
         terminate(threadPool);
     }
+
+    protected Comparator<Float> floatComparator() {
+        return FloatComparator.INSTANCE;
+    }
+
+    static final class FloatComparator implements Comparator<Float> {
+
+        static final FloatComparator INSTANCE = new FloatComparator();
+
+        @Override
+        public int compare(Float o1, Float o2) {
+            float first = o1;
+            float second = o2;
+            if (first < second) {
+                return -1;
+            } else if (first == second) {
+                return 0;
+            } else {
```

Comment:
I guess we can assume `o1` and `o2` to never be `NaN`?

Reply:
We never make NaN or +Inf or -Inf from out functions. Those always become `null`. Might be sane to be paranoid here and assert they aren't those things. Unsure, just drive by comment.

Reply:
Having seen where this is used in the tests, this is 100% fine. Never mind.